### ISSUE TITLE
Create method of automatically or dynamically re-sorting workflow outputs using tags

### ISSUE THREAD

**Issue by astrovsky01** [galaxyproject/galaxy#8374]
When a workflow is run, the outputs are a series of files as would output from each of the tools independently, save for those selected to be hidden. When a longer workflow runs on multiple input datasets, this can rapidly clutter the user history. I propose a method of automatic tag generation which can be used by the history panel to sort and dynamically re-sort the outputs of workflows into collections based on either 1) each individual tool's output in the workflow for all input datasets, or 2) the output of the workflow as a whole for each combination of input files

**Comment by astrovsky01** [comment]
My current plan looks as follows:

- On invocation of a workflow:
-     Add tag of all input file numbers:
-     Add tag of workflow name
- On each tool in a workflow:
-     Add tag for tool id + extra symbol

- Organization:
-     By tool in workflow:
-         For all outputs from workflow:
-             Filter tags list to [tag+symbol], take only the last element to deal with tag inheritance
-             Create collection for each unique tag+symbol
-             Output all files with that symbol to given collection
-             Tag all collections with the workflow tag to allow reorganization
-     By workflow input:
-         For all outputs from workflow:
-             Filter out tags with symbols
-             For all unique combinations of filtered tags:
-                 Generate a collection and populate it
-                 Label collections as [workflow x with inputs a,b,c]
-                 Tag collection as originating from workflows to allow reorganization.

- Re-sorting:
-     For all collections with tags:
-         Get current sort type
-         Un-list everything
-         Perform other sort type

**Comment by mvdbeek** [comment]
I don't think this is possible without a lot of large changes to the codebase and seems to also collide with ongoing work (I think, haven't seen any plans or WIPs but I heard that @Nerdinacan and @dannon are working on this) of either displaying the workflow invocations in the history panel or to have a separate view for workflow invocations.

I also don't think that persistent tags are the right way to do this, all the information in the tags you want to create is already stored in the database.

I'm not sure what exactly you mean by `collection`, If these are dataset collections as they exist today this doesn't seem like a good idea because they inform the flow of data and can't be reshaped dynamically. If you mean some container that summarizes outputs by workflow steps that seems more reasonable, and that would probably be a nice first step that can (probably) be implemented with the current state of the API.

**Comment by astrovsky01** [comment]
I was using the terminology of what currently is closest to what I want to do that exists, but by collection, I essentially just mean a file structure that is accessible in the history. The container you describe is more what I intend it to be. I had thought that using the current "operations on multiple datasets" could be a good foundation.

I had heard about the other work going on regarding the workflow invocation, and I was planning on getting in contact after posting the issue fo this idea. I agree, though, that the summary container is the first step.
